File: HackerRank/1 WeeK Preparation Kit/GridChallenge.py
def gridChallenge(grid):
    # Write your code here
    temp = []
    for info in grid:
        info = sorted(list(info))
        temp.append(info)
    result = transpose(temp)

    for case in result:
        if case != sorted(case):
            return "NO"
    return "YES"

def transpose(grid) :
    row, col = len(grid), len(grid[0])
    result = [[0 for row in range(row)] for col in range(col)]

    for i in range(row):
        for j in range(col):
            result[j][i] = grid[i][j]
    return result
    

  
grid = ['ebacd', 'fghij', 'olmkn', 'trpqs', 'xywuv']
print(gridChallenge(grid))

File: HackerRank/1 WeeK Preparation Kit/GridChallenge.cs
public class Program
{
    private static void Main(string[] args)
    {
    }

    private static string gridChallenge(List<string> grid)
    {
        var temp = new List<string>();
        foreach (var info in grid)
        {
            var infoList = info.ToList();
            infoList.Sort();
            var newString = String.Join("", infoList);
            temp.Add(newString);
        }

        var result = temp
                    .SelectMany(inner => inner.Select((item, index) => new { item, index }))
                    .GroupBy(i => i.index, i => i.item)
                    .Select(g => g.ToList())
                    .ToList();

        foreach (var info in result)
        {
            var check = info.OrderBy(x => x).ToList();
            var sorted = String.Join("", check);
            var original = String.Join("", info);

            if (original != sorted)
                return "NO";
        }

        return "YES";
    }
}